lib/vega_lite/data.ex

```diff
@@ -327,6 +327,328 @@ defmodule VegaLite.Data do
     end
   end
 
+  @doc """
+  Returns the specification of a polar plot of the input data.
+
+  `data` must be a list containing one of the following kinds of tuple:
```

Comment:
Unfortunately I think this format is a bit out of our specification. :( The goal here is to pass data as is, like an Explorer DF, and not previously formatted data. If we need to do a lot of pre-processing, it may not be the best candidate for this library at the moment. :(

Reply:
The tuple is so you can specify different mark types for each entry in the collection

Reply:
I think it's important to have the ability to have polar plots somewhere around here, because vega-lite (and I believe Vega itself) doesn't support this natively.

Reply:
Also, the first entry in the tuple is passed directly onto `VegaLite.data_from_values` in the end, if you specify `:radius_marks`, which is the only function that relies on getting the values for auto-scaling.

Reply:
I don't deny it is an important feature, but it is arguable if it should be part of this library or this module. This library provides bindings around VegaLite and additions should be arguably outside. When we talked about VegaLite.Data, we decided to still keep here because they are mostly small conveniences focused on the data,I don't think the proposed function is a small wrapper.

It also introduces a potential conflict. If VegaLite ever adds a polar plot, then we either need to break this function or have duplicates.

Reply:
I don't see how a joint histogram is a lower level of abstraction than a polar plot

Reply:
> The tuple is so you can specify different mark types for each entry in the collection

In any case, it is different from what data means everywhere in this module. So we would need to change this API to something closer to what is defined here.

Reply:
If we add an option to not plot the grid and just return the data layer, we could receive a single data+mark specification and for plotting multiple data onto the same grid you'd have to do a reduce outside of this.

A bit less friendly, but doable, and we can just show how to do it in the examples

Reply:
I have dropped some comments below. We would need some changes to the API and remove some features (such as inference based on data). But beware that we may still decide to not include this feature, so it is understandable if you would prefer to drop it. :)

Reply:
I'd rather only work on the refactors if the feature is going to be accepted in the end. Otherwise it would just be a waste of time.

Reply:
I can't know if it will be accepted at the end because I can't know how it will look like. Perhaps you have a better idea than I do.

If you can do the changes asked below (don't assume fields, don't traverse data), and still think it is useful, then in theory there are no blockers.